### DISCUSSION TITLE
Multiline Tabs?

### DISCUSSION THREAD

**Discussion by TillCity** [vitmalina/w2ui#2094] [Q&A]
I make extensive use of W2UI layout panels - the ease of setting up a panel layout with toolbars, menus, tabs and buttons has saved me a ton of work.  The one thing that I find somewhat annoying: when the number of tabs exceeds the available width W2UI shows scroll arrows to the right and/or left.  While this is OK it would be nicer if there was the option of getting the excess tabs to flow over into a second row.  Is there an easy way to do this?

**Comment by vitmalina** [comment]
No easy way to do this. The only thing I can think of is having two tab controls and if there is overflow, just move new ones into the second one. I will think about it.

**Reply by TillCity** [reply]
So it is possible to add multiple tab controls to a panel?  And if so, they will stack one below the other automatically?

**Reply by mpf82** [reply]
> So it is possible to add multiple tab controls to a panel?

No, you'd have to create multiple w2tabs manually.

> And if so, they will stack one below the other automatically?

No, you'd have to take care of that yourself.

As I understand it, Vitali will think about adding it to the feature request list, so it may or may not be possible in the future.

However, if you wanna give it a try and want to take a shot at implementing such a feature, feel free to create a PR, as long as it's backwards compatible.

Cheers =)
